app/models/event.rb

```diff
@@ -21,4 +25,25 @@ def upcoming_agenda_dates
     def upcoming_events
       Event.this_week_forward.sort_by(&:occurs_at)
     end
+
+    # TODO: Here is an example of a location that is not at the capitol complex:
+    #   {:field=> {
+    #     :"building-name"=>"Alpena Community College - Granum Theatre",
+    #     :"street-address"=>"665 Johnson Street",
+    #     :city=>"Alpena",
+    #     :state=>[
+    #       {:"postal-code"=>"MI"},
+    #       {:"state-fullname"=>"Michigan"}
+    #     ],
+    #     :zip=>"49707"}}
+    # For now, events that are not at the capitol complex are not formated and
+    # cached, so their `location` will be empty.
+    def cache_formatted_location
+      building_name = raw_location[:'capitol-complex'][:building]
+      room_name = raw_location[:'capitol-complex'][:room]
+      self.location = "Building: #{building_name}, Room: #{room_name}"
+    rescue NoMethodError
```

Comment:
If the only other edge case we've seen so far is presence of `capitol-complex` field, or presence of `field`, we can probably deal with these as 2 separate edge cases rather than rescuing the `NoMethodError`. That would help communicate that there are only (we think) 2 edge cases/formats.

These are the only 2 scenarios I've seen so far on the scrapes I've run but I will defer to you on this because I've only looked at a couple scrapes.